plugins/modules/idrac_diagnostics.py

```diff
@@ -0,0 +1,633 @@
+TIMEOUT_NEGATIVE_OR_ZERO_MSG = "The parameter `job_wait_timeout` value cannot be negative or zero."
+WAIT_TIMEOUT_MSG = "The job is not complete after {0} seconds."
+START_TIME = "The specified scheduled start time occurs in the past, " \
+             "provide a future time to schedule the start time."
+INVALID_TIME = "The specified date and time `{0}` to schedule the diagnostics is not valid. Enter a valid date and time."
+END_START_TIME = "The end time `{0}` to schedule the diagnostics must be greater than the start time `{1}`."
+
+IGNORE_CERTIFICATE_WARNING = {"off": "Off", "on": "On"}
+PROXY_SUPPORT = {"off": "Off", "default_proxy": "DefaultProxy", "parameters_proxy": "ParametersProxy"}
+PROXY_TYPE = {"http": "HTTP", "socks": "SOCKS"}
+
+
+class ExportDiagnostics:
+    STATUS_SUCCESS = [200, 202]
+
+    def __init__(self, idrac, module):
+        self.idrac = idrac
+        self.module = module
+
+    def execute(self, module):
+        try:
+            share_type = module.params.get('share_parameters').get('share_type')
+            export_diagnostics_url = self.__get_export_diagnostics_url(module)
+            job_status = {}
+            self.__check_file_extension(module)
+            if share_type == "local":
+                export_diagnostics_status = self.__export_diagnostics_local(module, export_diagnostics_url)
+            elif share_type in ["http", "https"]:
+                export_diagnostics_status = self.__export_diagnostics_http(module, export_diagnostics_url)
+                job_status = self.get_job_status(module, export_diagnostics_status)
+            elif share_type == "cifs":
+                export_diagnostics_status = self.__export_diagnostics_cifs(module, export_diagnostics_url)
+                job_status = self.get_job_status(module, export_diagnostics_status)
+            elif share_type == "nfs":
+                export_diagnostics_status = self.__export_diagnostics_nfs(module, export_diagnostics_url)
+                job_status = self.get_job_status(module, export_diagnostics_status)
+            status = export_diagnostics_status.status_code
+            if status in self.STATUS_SUCCESS:
+                msg = SUCCESS_EXPORT_MSG
+                job_details = job_status
+                return msg, job_details
+        except HTTPError as err:
+            err_msg = json.loads(err.read())["error"]["@Message.ExtendedInfo"][0]["Message"]
+            if err_msg == EXPORT_NO_RESULTS_MSG:
+                module.exit_json(msg=EXPORT_NO_RESULTS_MSG, skipped=True)
+            module.exit_json(msg=err_msg, failed=True)
+        return export_diagnostics_status
+
+    def __export_diagnostics_local(self, module, export_diagnostics_url):
+        payload = {}
+        payload["ShareType"] = "Local"
+        path = module.params.get('share_parameters').get('share_name')
+        if not (os.path.exists(path) or os.path.isdir(path)):
```

Comment:
if checking isdir(path), no need to do os.path.exists(path), doing duplicate work.

Reply:
done